internal/monitors/statsd/monitor.go

```diff
@@ -0,0 +1,177 @@
+package statsd
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/sfxclient"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "statsd"
+
+var metricTypeMap = map[string]datapoint.MetricType{
+	"g":  datapoint.Gauge,
+	"c":  datapoint.Count,
+	"ms": datapoint.Gauge,
+	"s":  datapoint.Gauge,
+}
+
+var logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 30*time.Second)
+var golibLogger = &utils.LogrusGolibShim{FieldLogger: logger.FieldLogger}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// converterInput is to receive configs to setup metric converters
+type converterInput struct {
```

Comment:
Make this exported.  It makes wrapping this a lot easier (which you will need for the AppMesh monitor).